pkg/operators/klusterlet/controllers/statuscontroller/klusterlet_status_controller.go

```diff
@@ -303,6 +303,19 @@ func buildKubeClientWithSecret(secret *corev1.Secret) (kubernetes.Interface, err
 	if err != nil {
 		return nil, err
 	}
+
+	// use embeded cert/key data instead of references to external cert/key files to
```

Comment:
do we still need line  291-304? if we read from secret directly, i think you could  build restConfig using  byte  directly by  using clientcmd.Load()

Reply:
Code refactored as suggested.